api/src/db/db.service.spec.ts

```diff
@@ -164,6 +164,59 @@ describe("DbService", () => {
         expect(docCount).toBe(8);
     });
 
+    it("can detect that a document is exactly the same as the document in the database", async () => {
+        const res: any = await service.upsertDoc({
+            _id: "group-private-content",
+            type: "group",
+            updatedTimeUtc: 3,
+            name: "Private Content",
+            acl: [
+                {
+                    type: "post",
+                    groupId: "group-private-users",
+                    permission: ["view"],
+                },
+                {
+                    type: "tag",
+                    groupId: "group-private-users",
+                    permission: ["view"],
+                },
+                {
+                    type: "post",
+                    groupId: "group-private-editors",
+                    permission: ["view", "edit", "translate", "publish"],
+                },
+                {
+                    type: "tag",
+                    groupId: "group-private-editors",
+                    permission: ["view", "translate", "assign"],
+                },
+                {
+                    type: "group",
+                    groupId: "group-private-editors",
+                    permission: ["view", "assign"],
+                },
+            ],
+        });
+
+        expect(res).toBe("passed document equal to existing database document");
+    });
+
+    it("can handle simultaneous updates to a single document", async () => {
```

Comment:
This test is failing in CI but works in local testing.

Gives the following error in CI:
<img width="740" alt="image" src="https://github.com/bccsa/ac-app/assets/13463323/67124314-a939-4ec9-8e0d-4d1ffebcfb34">

Reply:
Hmm weird since it seemingly complains about JSON.stringify and also about the socket which is not referenced in the db service 🤔

Reply:
Seems this error is related to first run with an empty database. On second run it passes.

Reply:
Solved. The issue was that the test which is testing simultaneous document updating ran on an empty database (while expecting the document to exist). Added a line to first create a document before doing the simultaneous update test.